src/Sleep.js

```diff
@@ -3,78 +3,77 @@ import sleepData from './data/sleep';
 class Sleep {
   constructor(sleepData) {
     this.sleepData = sleepData;
-  }
-  calculateAverageSleep(id) {
-    let perDaySleep = this.sleepData.filter((data) => id === data.userID);
-    return perDaySleep.reduce((sumSoFar, data) => {
-      return sumSoFar += data.hoursSlept;
-    }, 0) / perDaySleep.length;
-  }
-  calculateAverageSleepQuality(id) {
-    let perDaySleepQuality = this.sleepData.filter((data) => id === data.userID);
-    return perDaySleepQuality.reduce((sumSoFar, data) => {
-      return sumSoFar += data.sleepQuality;
-    }, 0) / perDaySleepQuality.length;
-  }
-  calculateDailySleep(id, date) {
-    let findSleepByDate = this.sleepData.find((data) => id === data.userID && date === data.date);
-    return findSleepByDate.hoursSlept;
-  }
-  calculateDailySleepQuality(id, date) {
+  };
+
+  calculateAverage(id, property) {
+    let dataPerDay = this.sleepData.filter((data) => id === data.userID);
+    let total = dataPerDay.reduce((sum, data) => {
+      return sum += data[property];
+    }, 0);
+    let averagePerDay = total / dataPerDay.length;
+    return averagePerDay;
+  };
+
+  calculateDailyTotal(id, date, property) {
     let findSleepQualityByDate = this.sleepData.find((data) => id === data.userID && date === data.date);
-    return findSleepQualityByDate.sleepQuality;
-  }
-  calculateWeekSleep(date, id, userRepo) {
-    return userRepo.getWeekFromDate(date, id, this.sleepData).map((data) => `${data.date}: ${data.hoursSlept}`);
-  }
-  calculateWeekSleepQuality(date, id, userRepo) {
-    return userRepo.getWeekFromDate(date, id, this.sleepData).map((data) => `${data.date}: ${data.sleepQuality}`);
-  }
+    return findSleepQualityByDate[property];
+  };
+
+  calculateWeekTotal(date, id, userRepo, property) {
+  return userRepo.getWeekFromDate(date, id, this.sleepData).map((data) => `${data.date}: ${data[property]}`);
+};
+
```

Comment:
Looked at this all together - Don't think we missed anything, but let's watch out to make sure these refactors to combine methods don't later affect anything unintentionally.